lib/Mojo/Promise.pm

```diff
@@ -288,6 +310,27 @@ rejects. If the returned promise fulfills, it is fulfilled with the values from
 the fulfilled promises in the same order as the passed promises. This method can
 be useful for aggregating results of multiple promises.
 
+=head2 all_settled
+
+  my $new = Mojo::Promise->all_settled(@promises);
+  $new->then(sub ($results) {
+    for my $result (@$results) {
+      if ($result->{status} eq 'fulfilled') {
+        say "fullfiled with $result->{value}";
+      } else {
+        say "rejected with $result->{reason}";
+      }
+    }
+  });
+
+Returns a new L<Mojo::Promise> object that fulfills when all of the
+passed L<Mojo::Promise> objects have either fulfilled or rejected.
+
+It is fulfilled with the values from the fulfilled promises and reasons from
+rejected promises in the same order as the passed promises. This method can
+be useful for aggregating results of multiple promises.
```

Comment:
This needs a better description, and an example that explains the resulting data structure. Ideally inspired by the Mozilla docs.

Reply:
Improve description.